2.0/m_lusersnoservices.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2008-2016 Naram Qashat <cyberbotx@cyberbotx.com>
+ *
+ * This file is a module for InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "inspircd.h"
+
+/* $ModDesc: Adds an extra line to LUSERS output to show global user count minus "clients" from U-Lines servers */
+/* $ModAuthor: Naram Qashat (CyberBotX) */
+/* $ModAuthorMail: cyberbotx@cyberbotx.com */
+/* $ModDepends: core 2.0 */
+
+class ModuleLusersNoServices : public Module
+{
+public:
+	void init()
+	{
+		ServerInstance->Modules->Attach(I_OnPreCommand, this);
+	}
+
+	void Prioritize()
+	{
+		Module *cmd_lusers = ServerInstance->Modules->Find("cmd_lusers");
+		if (cmd_lusers)
+			ServerInstance->Modules->SetPriority(this, I_OnPreCommand, PRIORITY_BEFORE, cmd_lusers);
+	}
+
+	Version GetVersion()
+	{
+		return Version("Adds an extra line to LUSERS output to show global user count minus \"clients\" from U-Lines servers");
+	}
+
+	ModResult OnPreCommand(std::string &command, std::vector<std::string> &parameters, LocalUser *user, bool validated, const std::string &original_line)
+	{
+		/* If the command doesnt appear to be valid, we dont want to mess with it. */
+		if (!validated)
+			return MOD_RES_PASSTHRU;
+
+		if (command == "LUSERS")
+		{
+			/* This tries to get the LUSERS of the cmd_lusers module output before ours */
+			Module *cmd_lusers = ServerInstance->Modules->Find("cmd_lusers");
+			if (cmd_lusers)
+				cmd_lusers->OnPreCommand(command, parameters, user, validated, original_line);
+			else
+			{
+				std::vector<std::string> emptyParameters;
+				ServerInstance->CallCommandHandler(command, emptyParameters, user);
+			}
+
+			/* Calculate how many clients are not psuedo-clients introduced by the Services package */
+			int ClientsNoServices = 0;
+			user_hash::const_iterator curr = ServerInstance->Users->clientlist->begin(), end = ServerInstance->Users->clientlist->end();
+			for (; curr != end; ++curr)
+				if (!ServerInstance->ULine(curr->second->server) && !ServerInstance->ULine(curr->second->nick))
```

Comment:
This is very performance intensive on large networks. You should calculate it once (on link?) and then cache it.